src/transformers/trainer.py

```diff
@@ -3192,9 +3190,13 @@ def evaluation_loop(
                 )
             if logits is not None:
                 logits = self._pad_across_processes(logits)
+            if self.preprocess_logits_for_metrics is not None and logits is not None:
+                logits = self.preprocess_logits_for_metrics(logits, labels)
+            if labels is not None:
+                labels = self._nested_gather(labels)
+                labels_host = labels if labels_host is None else nested_concat(labels_host, labels, padding_index=-100)
+            if logits is not None:
```

Comment:
Move the labels test after logits, so that the nested_gather line can be in the same `if logits is not None` as above.

Reply:
This is still pending. You can move line 3199 just after line 3194 and remove this new `if logits is not None` entirely.

Reply:
fix